Fix for ERA-5 API change

nate-kean opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
(Fixes #750)  
Modifies the fetch routine for ERA-5 to:
1. request lnsp/z separate from t/q, and
2. compute z for all model levels ourselves.

## How Has This Been Tested?
<!--- Please run the following command on your PR to ensure code formatting consistency: -->
<!--- autopep8 <changed files/folders> --recursive --in-place --ignore=E5 -->

<!--- Please describe how you tested your changes. -->


<!--- For new functionality, describe added unit tests. -->


<!--- If this PR breaks existing unit tests, please describe in detail -->
<!--- which tests break and why. Describe what functionality is changed. -->


## Screenshots (if appropriate):

## Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have added an explanation of what your changes do and why you'd like us to include them.
- [ ] I have written new tests for your core changes, as applicable.
- [x] I have successfully ran tests with your changes locally.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

nate-kean commented:
Here is a fix for the "model levels" level type. Untested for pressure levels and not incredibly efficient (there are some calculations that are being done multiple times unnecessarily), but I'm pushing this out due to the urgency of this issue. If this is merged in its current state I will file another pull request later with more tests and optimization.

jlmaurer commented on tools/RAiDER/delay.py:
```diff
@@ -73,8 +73,8 @@ def tropo_delay(
     # get heights
-    with xr.load_dataset(weather_model_file) as ds:
-        wm_levels = ds.z.values
+    with xr.open_dataset(weather_model_file) as ds:
```
@nate-kean is this a breaking change in xarray or just a different way to load data?

nate-kean replied:
This is actually mostly an optimization. `load_dataset` and `open_dataset` work the same, but `open` lazy loads the data, which in places where only a small part of a large dataset is being used, this can save a lot of memory.

jlmaurer commented on tools/RAiDER/interpolator.py:
```diff
@@ -156,13 +156,14 @@ def interpolate_elevation(dem_path: Union[Path, str], x: np.ndarray, y: np.ndarr
     Interpolates elevation values from a DEM to scattered points.
```
I'd recommend pulling out the files that are purely formatting issues and submitting those in a separate PR.

jlmaurer commented on tools/RAiDER/losreader.py:
```diff
@@ -71,6 +71,8 @@ def is_Projected(self):
     def ray_trace(self):
```
Same thing here. We can very quickly submit these as a separate PR just for formatting and typo changes.

jlmaurer commented on tools/RAiDER/models/era5t.py:
```diff
@@ -4,14 +4,14 @@
 
```
Same thing here in terms of formatting. you don't want to have too many lines of code changed in one PR.

nate-kean committed:
Update CHANGELOG.md for #751

nate-kean committed:
Rename warpedDEM.dem to .rdr

These are actually height files in RADAR coordinates, not DEMs in satellite coordinates. "A DEM that has been warped into a height file". This makes that more clear by giving them the file extension typically used for files in RADAR coordinates.

nate-kean committed:
Cherry-pick test_intersect.py from c1b6c21

nate-kean committed:
Update test_scenario_1

- Retooled to work with the current RAiDER API
- The test in test_HRRR_ztd.py had to be renamed to prevent a name collision

nate-kean committed:
Fix: Finish plumbing out dir arg to AOI classes

nate-kean committed:
Add/fix type annotations

Introduces a way to annotate the shape of numpy arrays, which can be extremely helpful for understanding and debugging the areas that do lots of work on arrays.

Add/fix type annotations

Add/fix type annotations

merge this into a delay.py commit

nate-kean committed:
Update for ERA-5 API changes

- Reworked height calculation for ECMWF's new method
- Reworked ECMWF._fetch() for their API changes: now makes two requests and calculates out the missing heights before saving
- Formatting on ECMWF._load_model_levels()

nate-kean committed:
Add ERA-5 to scenario_4 test

nate-kean committed:
Use raise-from for better debugging

jlmaurer reviewed (approved):
LGTM
